reframe/utility/__init__.py

```diff
@@ -54,6 +56,28 @@ def import_module_from_file(filename):
     return importlib.import_module(module_name)
 
 
+def allx(iterable):
+    """Return ``True`` if all elements of the `iterable` are true. If
+    iterable is empty or ``None`` return ``False``.
+
+    .. versionadded:: 2.13
+
+    """
+
+    # Check if iterable is an empty generator
+    if isinstance(iterable, types.GeneratorType):
+        try:
+            first_item = next(iterable)
+            return all(itertools.chain([first_item], iterable))
```

Comment:
Better put this in the `else` block of the try/except. Also rename `first_item` to `first` or `head`.